documentation/src/main/asciidoc/reference/search-dsl-highlighting.asciidoc

```diff
@@ -0,0 +1,473 @@
+----
+<1> Specify a predicate that would look for matches in both `title` and `description` fields.
+<2> Specify the details of the default highlighter. Setting the no match size to a positive value to let the highlighter know
+that we want to get some text back even if there will be nothing to highlight in a particular field.
+<3> Fetch the results.
+====
+
+[[search-dsl-highlighting-highlighter-type]]
+== Highlighter type
+
+Before a highlighter can be configured, you need to pick its type:
+
+Picking the highlighter type is the first step in a highlighter definition:
+
+.Specifying the highlighter type
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=various-highlighter-types]
+----
+<1> Starting the definition of a plain highlighter.
+<2> Starting the definition of a unified highlighter.
+<3> Starting the definition of a fast vector highlighter.
+====
+
+There are three options to choose from when it comes to the highlighter type:
+
+[[search-dsl-highlighting-highlighter-type-plain]]
+Plain::
+The plain highlighter can be useful for simple queries targeting a single field on a small number of documents.
+This highlighter uses a standard Lucene highlighter. It reads the string value of a highlighted field,
+then creates a small in-memory index from it and applies query logic to perform the highlighting.
+[[search-dsl-highlighting-highlighter-type-unified]]
+Unified::
+The unified highlighter is used by default and does not necessarily rely on re-analyzing the text, as it can
+get the offsets either from postings or from term vectors.
++
+This highlighter uses a break iterator (breaks the text into sentences by default) to break the text into later scored passages.
+It better supports more complex queries. Since it can work with prebuilt data, it performs better in case of a larger amount
+of documents compared to the plain highlighter.
+[[search-dsl-highlighting-highlighter-type-fast-vector]]
+Fast vector::
+In addition to using a break iterator similar to the unified highlighter, it can use the boundary characters
+to control the highlighted snippet.
++
+This is the only highlighter that can assign different weights to highlighted fragments, allowing it to show
+a fragment score differences by wrapping it with a different tag.
+For more on tags, see <<search-dsl-highlighting-highlighter-options-tags,the corresponding section>>.
+
+[NOTE]
+====
+If a full-text field that requires the `Highlightable.FAST_VECTOR` highlighter does not specify the term vector storage strategy,
+or set it to anything other than `TermVector.WITH_POSITIONS_OFFSETS`/`TermVector.WITH_POSITIONS_OFFSETS_PAYLOADS`,
+then the resulting term vector storing strategy for that field will get "upgraded" to one of `TermVector.WITH_POSITIONS_OFFSETS`/`TermVector.WITH_POSITIONS_OFFSETS_PAYLOADS`
+depending on the initially specified value.
+====
```

Comment:
Maybe this isn't the right place to talk about this? It seems more related to mapping, and I think you already explain this in the mapping section.

```suggestion
```

Alternatively, just mention in the description of the fast vector highlighter that enabling it requires mapping changes, and link to the appropriate section.